File: app/src/main/java/com/aaron/yespdf/filepicker/FileFilterImpl.java
package com.aaron.yespdf.filepicker;

import java.io.File;
import java.io.FileFilter;

/**
 * @author Aaron aaronzzxup@gmail.com
 */
class FileFilterImpl implements FileFilter {

    @Override
    public boolean accept(File file) {
        return !file.getName().startsWith(".") && (file.isDirectory() || file.getName().endsWith(".pdf"));
    }
}

File: app/src/main/java/com/aaron/yespdf/filepicker/FileFilterImpl.kt
package com.aaron.yespdf.filepicker

import java.io.File
import java.io.FileFilter

/**
 * @author Aaron aaronzzxup@gmail.com
 */
internal class FileFilterImpl : FileFilter {
    override fun accept(file: File): Boolean {
        return !file.name.startsWith(".") && (file.isDirectory || file.name.endsWith(".pdf"))
    }
}